src/vtp/web/api/backend.py

```diff
@@ -0,0 +1,219 @@
+found in that repo.  And that is due to two things: 1) the
+VTP-mock-election.US.xx holds the configuration of an election such as
+the blank ballot definition and 2) it also holds several hundred
+pre-cast random ballots so to fill the ballot cache so that
+ballot-checks can be immediately produced upon casting a ballot.
+
+Regardless, for the time being the VTP-mock-election.US.xx also holds
+checkedin mock values for the data that the web-api and above layers
+need when running in mock mode.
+"""
+
+import csv
+import json
+
+from vtp.core.common import Common
+from vtp.ops.accept_ballot_operation import AcceptBallotOperation
+from vtp.ops.cast_ballot_operation import CastBallotOperation
+from vtp.ops.setup_vtp_demo_operation import SetupVtpDemoOperation
+from vtp.ops.show_contests_operation import ShowContestsOperation
+from vtp.ops.tally_contests_operation import TallyContestsOperation
+from vtp.ops.verify_ballot_receipt_operation import VerifyBallotReceiptOperation
+
+
+class VtpBackend:
+    """Class to keep the namespace separate"""
+
+    ########
+    # backend demo constants
+    ########
+    # set mock mode
+    _MOCK_MODE = False
+    # where the blank ballot is stored for the spring demo
+    _MOCK_BLANK_BALLOT = "mock-data/blank-ballot.json"
+    # where the cast-ballot.json file is stored for the spring demo
+    _MOCK_CAST_BALLOT = "mock-data/cast-ballot.json"
+    # where the ballot-check is stored for the spring demo
+    _MOCK_BALLOT_CHECK = "mock-data/receipt.26.csv"
+    _MOCK_VOTER_INDEX = 26
+    # a mock contest content
+    _MOCK_CONTEST_CONTENT = "mock-data/mock_contest.json"
+    # default guid - making one up
+    _MOCK_GUID = "01d963fd74100ee3f36428740a8efd8afd781839"
+    # default mock receipt log
+    _MOCK_VERIFY_BALLOT_LOG = "mock-data/verify-ballot-doc.json"
+    # default mock tally log
+    _MOCK_TALLY_CONTESTS_LOG = "mock-data/tally-election-doc.json"
+    # default mock show contest log
+    _MOCK_SHOW_CONTEST_LOG = "mock-data/show-contest-doc.json"
+    # backend default address
+    _ADDRESS = "123, Main Street, Concord, Massachusetts"
+
+    @staticmethod
+    def get_vote_store_id() -> str:
+        """
+        Endpoint #1: will return a vote_store_id, a.k.a. a guid
+        """
+        if VtpBackend._MOCK_MODE:
+            # in mock mode there is no guid - make one up
+            return VtpBackend._MOCK_GUID
+        operation = SetupVtpDemoOperation(
+            election_data_dir=Common.get_generic_ro_edf_dir(),
+        )
+        return operation.run(guid_client_store=True)
+
+    @staticmethod
+    def get_empty_ballot(vote_store_id: str) -> dict:
+        """
+        Endpoint #2: given an existing guid, will return the blank
+        ballot
+        """
+        if VtpBackend._MOCK_MODE:
+            # in mock mode there is no guid - make one up
+            with open(VtpBackend._MOCK_BLANK_BALLOT, "r", encoding="utf8") as infile:
```

Comment:
Sure.  Note - I have yet to use that.  Given that AFAICT the VTP-web-api repo is not part of the demo and even if it is used for an expert to see, it will be on a git installed laptop, I am ok not addressing this at that time.  My current plan is that if there are any deep demo questions, I (or you or Neil) would use the CLI to demo it that to the deeper geek.

?

Reply:
Works for me. I said "eventually" because I realize it's not in scope yet.